pkg/catalog/catalog.go

```diff
@@ -107,9 +107,10 @@ const (
 )
 
 type Config struct {
-	Config        *config.Config
-	KVStore       *kv.StoreMessage
-	WalkerFactory WalkerFactory
+	Config             *config.Config
+	KVStore            *kv.StoreMessage
+	WalkerFactory      WalkerFactory
+	UseRepositoryCache bool
```

Comment:
Why is this a configurable value? Where do you set it to true (other than tests)

Reply:
Added comment explaining why we have a catalog level configuration that configure the ref manager's level.

Reply:
Why are we not just passing a cache object to the ref manager? That is our convention for managers for the most part

Reply:
At this level for simplicity as we are going to add more caches, enabling it using boolean looks like a good first step.
Next we can configure each one using application level settings.